api/v6/api.go

```diff
@@ -34,6 +35,7 @@ type ControllerStatus struct {
 	Containers []Container
 	ReadOnly   ReadOnlyReason
 	Status     string
+	Pods       cluster.Pods
```

Comment:
There doesn't seem much point in compiling a list of problems with the controller if it's not communicated in the API. May I suggest that the `ControllerStatus` has a field for rollout status which includes both the counts of pods and the reason it's stuck or still in progress, if it's not completed.